src/InsertionsClient.Console/ConsoleApp/InputLoading.cs

```diff
@@ -0,0 +1,74 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.Net.Insertions.ConsoleApp
+{
+    internal static class InputLoading
+    {
+        internal static void ProcessArgument(string argument, string appSwitch, string cmdLineMessage, out string target)
+        {
+            target = argument.Replace(appSwitch, string.Empty);
+            Trace.WriteLine($"CMD line param. {cmdLineMessage} {target}");
+        }
+
+        internal static void ProcessArgumentInt(string argument, string appSwitch, string cmdLineMessage, out int target)
+        {
+            string trimmedArg = argument.Replace(appSwitch, string.Empty);
+            if (int.TryParse(trimmedArg, out target))
+            {
+                Trace.WriteLine($"CMD line param. {cmdLineMessage} {target}");
+            }
+            else
+            {
+                target = -1;
+                Trace.WriteLine("Specified value is not an integer. Default value will be used.");
+            }
+        }
+
+        internal static IEnumerable<Regex> LoadWhitelistedPackages(string whitelistedPackagesFile)
+        {
+            if (!File.Exists(whitelistedPackagesFile))
+            {
+                return Enumerable.Empty<Regex>();
+            }
+
+            try
+            {
+                return File.ReadAllLines(whitelistedPackagesFile)
```

Comment:
minor: it the `.ToList()` needed?

Reply:
If we don't evaluate the results into a List now, new Regex objects will be allocated every time we iterate over the enumerable. This way, we do the allocation just once.